src/MongoDB/Driver/ReadPreference.cpp

```diff
@@ -67,11 +65,13 @@ Array HHVM_METHOD(MongoDBDriverReadPreference, __debugInfo)
 	MongoDBDriverReadPreferenceData* data = Native::data<MongoDBDriverReadPreferenceData>(this_);
 	Array retval = Array::Create();
 	Variant v_tags;
+	const bson_t *tags;
 
-	retval.set(s_mode, data->m_read_preference->mode);
+	retval.set(s_mode, mongoc_read_prefs_get_mode(data->m_read_preference));
 
 	hippo_bson_conversion_options_t options = HIPPO_TYPEMAP_DEBUG_INITIALIZER;
-	BsonToVariantConverter convertor(bson_get_data(&data->m_read_preference->tags), data->m_read_preference->tags.len, options);
+	tags = mongoc_read_prefs_get_tags(data->m_read_preference);
```

Comment:
Any reason this isn't consolidated into the declaration, as you did in `MongoDBDriverCursor::__debugInfo`?